src/net.h

```diff
@@ -1002,6 +1004,44 @@ class CConnman
     mapMsgTypeSize m_msgtype_bytes_recv;
     mapMsgTypeSize m_msgtype_bytes_sent;
 
+    struct NetMsgStatsKey {
+        // protocol.cpp has a list of allNetMessageTypes which use the NetMsgType namespace.
+        // Can I treat NetMsgType like an enum?
+        std::string msg_type;
```

Comment:
Revisit this

Reply:
Good point about being able to reuse things like the `mapMsgTypeSize` alias! Is that what you were referring to when you mentioned `CNodeStats`? I didn't see any `std::string` message type fields on that class. Just want to make sure I'm not missing something. 

`CNodeStats` has two members of type `mapMsgTypeSize`. I assume this is what you were referring to as an example of existing patterns that use message type strings?

Regarding the second half of your comment, at some point you threw out the idea of turning the message types into an enum, but as a separate PR that is completely decoupled from this. I can't recall if the comment I'm replying to here was before or after that discussion. Can you remind me what the latest decision was?

Reply:
This comment is no longer relevant now that the workaround for the message type enum is in place (yay!). Closing this comment.